apps/prairielearn/src/lib/client/page-context.ts

```diff
@@ -0,0 +1,76 @@
+import { z } from 'zod';
+
+import {
+  type StaffCourse,
+  type StaffCourseInstance,
+  StaffCourseInstanceSchema,
+  StaffCourseSchema,
+  type StudentCourse,
+  type StudentCourseInstance,
+  StudentCourseInstanceSchema,
+  StudentCourseSchema,
+} from '../db-types.js';
+
+const PageContext = z.object({
+  authz_data: z.object({
+    has_course_instance_permission_edit: z.boolean(),
+    has_course_instance_permission_view: z.boolean(),
```

Comment:
These won't be defined on non-course-instance pages. Thankfully things will error hard when they're missing, so we'll at least be forced to handle this once it matters.

Not requesting a change here, just brainstorming: we may want more specific types like `CourseInstancePageContext` and `CoursePageContext`? Again, we can see how things shake out as we use this on more pages 🙂